x-pack/auditbeat/module/system/login/utmp.go

```diff
@@ -274,10 +280,39 @@ func (r *UtmpFileReader) updateSavedUtmpFile(utmpFile UtmpFile, f *os.File) erro
 	return nil
 }
 
-// processLoginRecord receives UTMP login records in order and returns
+// processBadLoginRecord takes a UTMP login record from the "bad" login file (/var/log/btmp)
+// and returns a LoginRecord for it.
+func (r *UtmpFileReader) processBadLoginRecord(utmp *Utmp) (*LoginRecord, error) {
+	record := LoginRecord{
+		Utmp:      utmp,
+		Timestamp: utmp.UtTv,
+		TTY:       utmp.UtLine,
+		UID:       -1,
+		PID:       -1,
+	}
+
+	switch utmp.UtType {
+	// See utmp(5) for C constants.
+	case LOGIN_PROCESS, USER_PROCESS:
+		record.Type = userLoginFailedRecord
+
+		record.Username = utmp.UtUser
+		record.UID = lookupUsername(record.Username)
+		record.PID = utmp.UtPid
+		record.IP = newIP(utmp.UtAddrV6)
+		record.Hostname = utmp.UtHost
+	default:
+		// This should not happen.
+		return nil, errors.Errorf("UTMP record with unexpected type %v in bad login file.", utmp.UtType)
```

Comment:
Don't end the error with punctuation just in case it gets wrapped by something else. https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Reply:
Thanks, fixed.